api/src/main/java/org/openmrs/Obs.java

```diff
@@ -1175,93 +1169,15 @@ public void setDateCreated(Date dateCreated) {
 		markAsDirty(getDateCreated(), dateCreated);
 		super.setDateCreated(dateCreated);
 	}
-	
-	/**
-	 * Gets the namespace for the form field that was used to capture the obs details in the form
-	 * 
-	 * @return the namespace
-	 * @since 1.11
-	 * @should return the namespace for a form field that has no path
-	 * @should return the correct namespace for a form field with a path
-	 * @should return null if the namespace is not specified
-	 */
-	public String getFormFieldNamespace() {
-		if (StringUtils.isNotBlank(formNamespaceAndPath)) {
-			//Only the path was specified
-			if (formNamespaceAndPath.startsWith(FORM_NAMESPACE_PATH_SEPARATOR)) {
-				return null;
-			}
-			return formNamespaceAndPath.substring(0, formNamespaceAndPath.indexOf(FORM_NAMESPACE_PATH_SEPARATOR));
-		}
-		
-		return formNamespaceAndPath;
-	}
-	
-	/**
-	 * Gets the path for the form field that was used to capture the obs details in the form
-	 * 
-	 * @return the the form field path
-	 * @since 1.11
-	 * @should return the path for a form field that has no namespace
-	 * @should return the correct path for a form field with a namespace
-	 * @should return null if the path is not specified
-	 */
-	public String getFormFieldPath() {
-		if (StringUtils.isNotBlank(formNamespaceAndPath)) {
-			//Only the namespace was specified
-			if (formNamespaceAndPath.endsWith(FORM_NAMESPACE_PATH_SEPARATOR)) {
-				return null;
-			}
-			return formNamespaceAndPath.substring(formNamespaceAndPath.indexOf(FORM_NAMESPACE_PATH_SEPARATOR) + 1);
-		}
-		
-		return formNamespaceAndPath;
-	}
-	
+
 	/**
-	 * Sets the namespace and path of the form field that was used to capture the obs details in the
-	 * form.<br>
-	 * <b>Note:</b> Namespace and formFieldPath together must not exceed 254 characters in length,
-	 * form applications can subtract the length of their namespace from 254 to determine the
-	 * maximum length they can use for a form field path.
-	 * 
-	 * @param namespace the namespace of the form field
-	 * @param formFieldPath the path of the form field
-	 * @since 1.11
-	 * @should set the underlying formNamespaceAndPath in the correct pattern
-	 * @should reject a namepace containing the separator
-	 * @should reject a path containing the separator
-	 * @should reject a namepace and path combination longer than the max length
-	 * @should not mark the obs as dirty when the value has not been changed
-	 * @should mark the obs as dirty when the value has been changed
-	 * @should mark the obs as dirty when the value is changed from a null to a non null value
-	 * @should mark the obs as dirty when the value is changed from a non null to a null value
+	 * @see org.openmrs.FormRecordable#setFormField(String, String)
 	 */
+	@Override
 	public void setFormField(String namespace, String formFieldPath) {
-		if (namespace == null && formFieldPath == null) {
-			markAsDirty(formNamespaceAndPath, null);
-			formNamespaceAndPath = null;
-			return;
-		}
```

Comment:
What about this if statement, didn't it get wiped through the changes?

Reply:
This if statement, in on  `BaseFormRecordableOpenmrsData.setFormField` same as version `2.4`

Reply:
But not that call:
```java
markAsDirty(formNamespaceAndPath, null);
```
This seems to have been removed through streamlining the code between the base class and `Obs` no?

Reply:
Maybe we can do a discuses this during the meeting tomorrow. 
But markAsDirty will receive the oldValue amd the new value. And I make sure that this still true by saving the oldValue before call the `super.setFormField(namespace, formFieldPath)` and call the `markAsDirty` at the end with the olde value that I saved and the new one calculate in the `setFormField`.
```
private void markAsDirty(Object oldValue, Object newValue) {
```